test/libsolidity/smtCheckerTests/external_calls/external_reentrancy_crypto.sol

```diff
@@ -0,0 +1,33 @@
+pragma experimental SMTChecker;
+
+abstract contract D {
+	function d() virtual public;
+}
+
+contract C {
+	bytes data;
+
+	bytes32 kec;
+
+	constructor(bytes memory _data) {
+		data = _data;
+
+		kec = keccak256(data);
+	}
+
+	function check(bytes memory _data) public view {
+		bytes32 _kec = keccak256(data);
+		assert(_kec == kec); // should hold
+		assert(kec == keccak256(_data)); // should fail
+	}
+
+	function ext(D d) public {
+		d.d();
+	}
+}
+// ----
+// Warning 1218: (335-366): CHC: Error trying to invoke SMT solver.
+// Warning 6328: (335-366): CHC: Assertion violation might happen here.
+// Warning 1218: (335-366): CHC: Error trying to invoke SMT solver.
+// Warning 6328: (335-366): CHC: Assertion violation might happen here.
```

Comment:
Looks like CHC is running the assertion twice?

Reply:
@blishko Is this because the assertion can be violated both via `ext` and `check`?

Reply:
Yes, I believe that's the reason.